scripts/update-screenshots.js

```diff
@@ -106,69 +96,85 @@ const updateScreenshots = async (config) => {
 
           break;
         case 'Filament':
-          const {width, height} = scenario.dimensions;
-
-          if (modelSource == null) {
-            warn(`Could not determine model source for ${
-                scenario.slug}; skipping...`);
-            continue;
-          }
-
-          if (backgroundImage == null) {
-            warn(`Could not determine IBL for ${scenario.slug}; skipping...`);
-            continue;
-          }
-
-          const backgroundImagePath =
-              path.resolve(path.dirname(testHtmlPath), backgroundImage);
-
-          const modelSourcePath =
-              path.resolve(path.dirname(testHtmlPath), modelSource);
-
-          await new Promise((resolve, reject) => {
-            console.log(`🖌️  Rendering ${name} screenshot for ${slug}...`);
-
-            const childProcess = spawn(
-                filamentScreenshotScript,
-                [
-                  '-w',
-                  `${width}`,
-                  '-h',
-                  `${height}`,
-                  '-i',
-                  backgroundImagePath,
-                  '-m',
-                  modelSourcePath,
-                  '-o',
-                  filePath
-                ],
-                {
-                  cwd: process.cwd(),
-                  env: process.env,
-                  stdio: ['ignore', 'inherit', 'inherit']
-                });
-
-            childProcess.once('error', (error) => {
-              warn(error);
-            });
-
-            childProcess.once('exit', (code) => {
-              if (code === 0) {
-                console.log(
-                    `✅ Successfully captured screenshot for ${name} ${slug}`);
-                resolve();
-              } else {
-                reject(new Error('Failed to capture Filament screenshot'));
-              }
-            });
-          });
-
+          await screenshotFromScript(scenario, scenarioDirectory, filePath, name, filamentScreenshotScript);
           break;
       }
     }
   }
 };
 
+const screenshotFromScript = async (scenario, scenarioDirectory, filePath, name, script) => {
+  const testHtmlPath = path.join(scenarioDirectory, 'index.html');
+
+  const html = (await fs.readFile(testHtmlPath)).toString();
+
+  const backgroundImageMatch = html.match(backgroundImageRe);
+  const backgroundImage =
```

Comment:
The original implementation extracts these test configuration values in the least good way that actually works (RegExp parsing attributes from HTML strings).

It might make more sense to parse the HTML strings once outside of this function and pass the extracted values in here as arguments.

Reply:
I'm going to give this a pass (in a separate CL). I agree, either this needs to be cleaned up, but I wonder if we should switch to a JSON storage format to clean up adding both new renderers and new models.

Reply:
Yes, and yes.

Reply:
Sorry, but also, yes, but maybe later to save time. Your call :)